tools/shared/src/main/scala/org/scalajs/core/tools/linker/checker/IRChecker.scala

```diff
@@ -301,7 +301,7 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
       }
     }
 
-    if (classDef.kind.isJSClass && name == "constructor") {
+    if (classDef.kind.isJSClass && name == "constructor" && !static) {
```

Comment:
This makes me think: What happens if I export a static under the name "constructor"?